File: ros2_ws/src/ros2_app_py/ros2_app_py/number_publisher.py
#!/usr/bin/env pytohn3
import rclpy
from rclpy.node import Node
from std_msgs.msg import Int8


class NumberPublisherNode(Node):
    def __init__(self):
        super().__init__("number_publisher")

        # Declare parameters.
        self.declare_parameter("pub_freq", value=1.0)
        self.declare_parameter("num2pub", value=0)

        # Get value from parameters.
        self.pub_freq_ = self.get_parameter("pub_freq").value
        self.num2pub_ = self.get_parameter("num2pub").value

        self.number_timer_ = self.create_timer(
            1.0 / self.pub_freq_, self.publish_number)
        self.pub_ = self.create_publisher(Int8, 'number', 10)

        # Confirm node initialization.
        self.get_logger().info("number_publisher Python node initialized.")

    def publish_number(self):
        msg = Int8()
        msg.data = self.num2pub_
        self.pub_.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: ros2_ws/src/ros2_app_cpp/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/int8.hpp"

class NumberPublisherNode : public rclcpp::Node
{
public:
    NumberPublisherNode() : Node("number_publisher")
    {
        RCLCPP_INFO(this->get_logger(), "number_publisher C++ node initialized.");
        this->declare_parameter("pub_freq", 1.0);
        this->declare_parameter("num2pub", 0);

        number_ = this->get_parameter("num2pub").as_int();
        pub_freq_ = this->get_parameter("pub_freq").as_double();

        number_timer_ = this->create_wall_timer(std::chrono::milliseconds(int(1000.0 / pub_freq_)),
                                                std::bind(&NumberPublisherNode::publish_number, this));

        pub_ = this->create_publisher<std_msgs::msg::Int8>("number", 10);
    }

private:
    void publish_number()
    {
        auto msg = std_msgs::msg::Int8();
        msg.data = number_;
        pub_->publish(msg);
    }

    int number_;
    double pub_freq_;
    rclcpp::Publisher<std_msgs::msg::Int8>::SharedPtr pub_;
    rclcpp::TimerBase::SharedPtr number_timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberPublisherNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}